src/include/duckdb/main/query_profiler.hpp

```diff
@@ -89,32 +112,44 @@ class OperatorProfiler {
 	Profiler<system_clock> op;
 	//! The stack of Physical Operators that are currently active
 	std::stack<PhysicalOperator *> execution_stack;
-	//! A mapping of physical operators to recorded timings
-	unordered_map<PhysicalOperator *, OperatorTimingInformation> timings;
+	//! A mapping of physical operators to operator information
+	unordered_map<PhysicalOperator *, OperatorInformation> timings;
 };
 
 //! The QueryProfiler can be used to measure timings of queries
 class QueryProfiler {
+public:
+	DUCKDB_API QueryProfiler()
+	    : automatic_print_format(ProfilerPrintFormat::NONE), enabled(false), detailed_enabled(false), running(false),
+	      query_requires_profiling(false) {
+	}
+
 public:
 	struct TreeNode {
 		string name;
 		string extra_info;
-		OperatorTimingInformation info;
+		OperatorInformation info;
 		vector<unique_ptr<TreeNode>> children;
 		idx_t depth = 0;
 	};
 
+	//! The lock used for flushing information from a thread into the global query profiler
+	mutex flush_lock;
```

Comment:
A public lock is usually asking for trouble. Can this be made private, and obtained by e.g. the Flush method?